### PR TITLE
Removed use of .tag and Renamed ViewPager

### PR DESCRIPTION
**Pull request by BatmanInGotham2**

### CODE REVIEW

File: app/src/main/java/com/google/samples/apps/sunflower/HomeViewPagerFragment.kt [hunk 1]
```diff
@@ -61,4 +61,13 @@ class ViewPagerFragment : Fragment() {
 
         return binding.root
     }
+
+    /** Return the list of tab titles to pass to SunflowerPagerAdapter **/
+    private fun getTabTitles(): List<String> {
+        val tabTitles: MutableList<String> = arrayListOf()
```
Comment (tiembo) [inline app/src/main/java/com/google/samples/apps/sunflower/HomeViewPagerFragment.kt]: What do you think about using a HashMap?  That way the case statement in `getPageTitle` can be simplified to `tabTitles.get(position)`?

Comment (BatmanInGotham2) [reply app/src/main/java/com/google/samples/apps/sunflower/HomeViewPagerFragment.kt]: I agree, it would make for faster indexing too

Comment (BatmanInGotham2) [reply app/src/main/java/com/google/samples/apps/sunflower/HomeViewPagerFragment.kt]: updated

File: app/src/main/java/com/google/samples/apps/sunflower/adapters/SunflowerPagerAdapter.kt [hunk 2]
```diff
@@ -38,18 +38,10 @@ class SunflowerPagerAdapter(
...
+        return tabFragments[position] ?: throw IndexOutOfBoundsException()
     }
 
     override fun getPageTitle(position: Int): CharSequence? {
-        return when (position) {
-            MY_GARDEN_PAGE_INDEX -> context.getString(R.string.my_garden_title)
-            PLANT_LIST_PAGE_INDEX -> context.getString(R.string.plant_list_title)
-            else -> null
-        }
+        return tabTitles[position] ?: throw IndexOutOfBoundsException()
```
Comment (tiembo) [inline app/src/main/java/com/google/samples/apps/sunflower/adapters/SunflowerPagerAdapter.kt]: OK to return null here, so just `tabTitles[position]` works

File: app/src/main/java/com/google/samples/apps/sunflower/HomeViewPagerFragment.kt [hunk 3]
```diff
@@ -61,4 +61,16 @@ class ViewPagerFragment : Fragment() {
 
         return binding.root
     }
+
+    /** Return a hashmap from tab index to tab title to pass to SunflowerPagerAdapter **/
+    private fun getTabTitles(): HashMap<Int, String> {
```
Comment (XinyueZ) [inline app/src/main/java/com/google/samples/apps/sunflower/HomeViewPagerFragment.kt]: `SparseArrayCompat` would say

**Pull request merged**